packages/middleware-flexible-checksums/src/resolveFlexibleChecksumsConfig.ts

```diff
@@ -0,0 +1,41 @@
+import { Encoder, HashConstructor, StreamHasher } from "@aws-sdk/types";
+
+export interface FlexibleChecksumsInputConfig {}
+
+interface PreviouslyResolved {
+  /**
+   * The function that will be used to convert binary data to a base64-encoded string.
+   * @internal
+   */
+  base64Encoder: Encoder;
+
+  /**
+   * A constructor for a class implementing the {@link Hash} interface that computes MD5 hashes.
+   * @internal
+   */
+  md5: HashConstructor;
+
+  /**
+   * A constructor for a class implementing the {@link Hash} interface that computes SHA1 hashes.
+   * @internal
+   */
+  sha1: HashConstructor;
+
+  /**
+   * A constructor for a class implementing the {@link Hash} interface that computes SHA256 hashes.
+   * @internal
+   */
+  sha256: HashConstructor;
+
+  /**
+   * A function that, given a hash constructor and a stream, calculates the hash of the streamed value.
+   * @internal
+   */
+  streamHasher: StreamHasher<any>;
+}
+
+export interface FlexibleChecksumsResolvedConfig extends PreviouslyResolved {}
+
+export const resolveFlexibleChecksumsConfig = <T>(
+  input: T & PreviouslyResolved & FlexibleChecksumsInputConfig
+): T & FlexibleChecksumsResolvedConfig => input;
```

Comment:
I don't think it's needed. It seems like this plugin doesn't require any new client configuration. So you only need middleware to do the job. None of `resolveFlexibleChecksumsConfig()` or `FlexibleChecksumsResolvedConfig` is needed for export.

Reply:
Good catch! It was added in the initial implementation.
Removed it in [`df642fa` (#3340)](https://github.com/aws/aws-sdk-js-v3/pull/3340/commits/df642fa23dc8c9579d5ebb3daa72f2f9cdf27a49)